biojava-core/src/main/java/org/biojava/nbio/core/sequence/io/ABITrace.java

```diff
@@ -0,0 +1,568 @@
+                    if (i == 2) G[x] = (int) dis.readShort();
+                    if (i == 3) T[x] = (int) dis.readShort();
+                } catch (IOException e)//This shouldn't happen. If it does something must be seriously wrong.
+                {
+                    throw new IllegalStateException("Unexpected IOException encountered while manipulating internal streams.");
+                }
+            }
+        }
+        return;
+    }
+
+    /**
+     * Fetch the sequence from the trace data.
+     */
+    private void setSeq() {
+        char tempseq[] = new char[SeqLength];
+        for (int x = 0; x <= SeqLength - 1; ++x) {
+            tempseq[x] = (char) TraceData[PBAS2 + x];
+        }
+        sequence = new String(tempseq);
+    }
+
+    /**
+     * Fetch the quality calls from the trace data.
+     */
+    private void setQcalls() {
+        Qcalls = new int[SeqLength];
+        byte[] qq = new byte[SeqLength];
+        getSubArray(qq, PCON);
+        DataInputStream dis = new DataInputStream(new ByteArrayInputStream(qq));
+        for (int i = 0; i <= SeqLength - 1; ++i) {
+            try {
+                Qcalls[i] = (int) dis.readByte();
+            } catch (IOException e)//This shouldn't happen. If it does something must be seriously wrong.
+            {
+                throw new IllegalStateException("Unexpected IOException encountered while manipulating internal streams.");
+            }
+        }
+    }
+
+    /**
+     * Fetch the basecalls from the trace data.
+     */
+    private void setBasecalls() {
+        Basecalls = new int[SeqLength];
+        byte[] qq = new byte[SeqLength * 2];
+        getSubArray(qq, PLOC);
+        DataInputStream dis = new DataInputStream(new ByteArrayInputStream(qq));
+        for (int i = 0; i <= SeqLength - 1; ++i) {
+            try {
+                Basecalls[i] = (int) dis.readShort();
+            } catch (IOException e)//This shouldn't happen. If it does something must be seriously wrong.
+            {
+                throw new IllegalStateException("Unexpected IOException encountered while manipulating internal streams.");
+            }
+        }
+    }
+
+    /**
+     * Sets up all of the initial pointers to the important records in TraceData.
+     */
+    private void setIndex() {
+        int DataCounter, PBASCounter, PLOCCounter, PCONCounter, NumRecords;
+        byte[] RecNameArray = new byte[4];
+        String RecName;
+
+        DataCounter = 0;
+        PBASCounter = 0;
+        PLOCCounter = 0;
+        PCONCounter = 0;
+
+        IndexBase = getIntAt(AbsIndexBase + MacJunk);
+        NumRecords = getIntAt(AbsIndexBase - 8 + MacJunk);
+
+        for (int record = 0; record <= NumRecords - 1; record++) {
+            getSubArray(RecNameArray, (IndexBase + (record * 28)));
+            RecName = new String(RecNameArray);
+            if (RecName.equals("FWO_"))
+                FWO = IndexBase + (record * 28) + 20;
```

Comment:
Any idea what this 2 "magic" numbers mean? It'd be nice to extract them to constants with some documentation as to what they mean.

Reply:
I don't know what these 2 numbers stand for. What would be the best names to call them?

Reply:
I'm not familiar with the format either. They seem some kind of offset, but I can't really tell.

FYI the format is described http://www6.appliedbiosystems.com/support/software_community/ABIF_File_Format.pdf

Let's leave it as is then... unless there's other suggestions, @el1jaC ?